SNESDeltaCore/Bridge/SNESEmulatorBridge.mm

```diff
@@ -113,19 +113,25 @@ - (void)startWithGameURL:(NSURL *)URL
+    S9xSetController(1, CTL_JOYPAD, 1, 0, 0, 0);
     
-    S9xMapButton(SNESGameInputUp, S9xGetCommandT("Joypad1 Up"), NO);
-    S9xMapButton(SNESGameInputDown, S9xGetCommandT("Joypad1 Down"), NO);
-    S9xMapButton(SNESGameInputLeft, S9xGetCommandT("Joypad1 Left"), NO);
-    S9xMapButton(SNESGameInputRight, S9xGetCommandT("Joypad1 Right"), NO);
-    S9xMapButton(SNESGameInputA, S9xGetCommandT("Joypad1 A"), NO);
-    S9xMapButton(SNESGameInputB, S9xGetCommandT("Joypad1 B"), NO);
-    S9xMapButton(SNESGameInputX, S9xGetCommandT("Joypad1 X"), NO);
-    S9xMapButton(SNESGameInputY, S9xGetCommandT("Joypad1 Y"), NO);
-    S9xMapButton(SNESGameInputL, S9xGetCommandT("Joypad1 L"), NO);
-    S9xMapButton(SNESGameInputR, S9xGetCommandT("Joypad1 R"), NO);
-    S9xMapButton(SNESGameInputStart, S9xGetCommandT("Joypad1 Start"), NO);
-    S9xMapButton(SNESGameInputSelect, S9xGetCommandT("Joypad1 Select"), NO);
+    for (int player = 1; player <= 8; player++)
+    {
+        NSUInteger mask = player << 16;
+        
+        S9xMapButton(mask | SNESGameInputUp, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Up"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputDown, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Down"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputLeft, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Left"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputRight, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Right"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputA, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"A"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputB, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"B"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputX, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"X"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputY, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Y"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputL, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"L"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputR, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"R"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputStart, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Start"] UTF8String]), NO);
+        S9xMapButton(mask | SNESGameInputSelect, S9xGetCommandT([[[NSString stringWithFormat:@"Joypad%d ", player] stringByAppendingString:@"Select"] UTF8String]), NO);
+    }
```

Comment:
A little hard to parse at first glance, something like this would be easier to read for me:

```suggestion
    for (int player = 1; player <= 8; player++)
    {
        NSUInteger mask = player << 16;
        
        void (^mapInput)(uint32_t, NSString *) = ^(uint32_t input, NSString *inputName) {
            NSString *command = [NSString stringWithFormat:@"Joypad%d %@", player, inputName];
            S9xMapButton(mask | input, S9xGetCommandT(command.UTF8String), NO);
        };
        
        mapInput(SNESGameInputUp, @"Up");
        mapInput(SNESGameInputDown, @"Down");
        mapInput(SNESGameInputLeft, @"Left");
        mapInput(SNESGameInputRight, @"Right");
        mapInput(SNESGameInputA, @"A");
        mapInput(SNESGameInputB, @"B");
        mapInput(SNESGameInputX, @"X");
        mapInput(SNESGameInputY, @"Y");
        mapInput(SNESGameInputL, @"L");
        mapInput(SNESGameInputR, @"R");
        mapInput(SNESGameInputStart, @"Start");
        mapInput(SNESGameInputSelect, @"Select");
    }
```